react/react-ecosystem/type-checking-with-proptypes.md

```diff
@@ -0,0 +1,100 @@
+### Introduction
+
+PropTypes is a way to type check the props that a React component receives. It helps to catch potential type errors during development, making it easier to spot and fix bugs. If you have used a linter in your previous React projects, there's a good chance it ended up yelling at you about certain props missing in prop validation, however if that isn't the case- well, lucky you!
+
+### Lesson Overview
+
+This section contains a general overview of topics that you will learn in this lesson.
+
+- Setting up PropTypes.
+- Using common PropTypes features.
+
+### Getting Started
+
+To start using PropTypes in our react projects, we first need to install the corresponding library. We can do that with ~~~npm~~~. In your react project run the following command:
+
+~~~
+npm install --save prop-types
+~~~
+
+Next, we want to import the PropTypes package in the component whose props we want to validate.
+
+~~~javascript
+import PropTypes from 'prop-types';
+~~~
+
+### Using propTypes
+
+Here is a very basic example of how we would use it in a simple component that renders out a name prop.
+
+~~~javascript
+import PropTypes from 'prop-types';
+import React from 'react';
+
+const RenderName = (props) => {
+  return <div>{props.name}</div>;
+};
+
+RenderName.propTypes = {
+  name: PropTypes.string,
+};
+
+export default RenderName;
+~~~
+
+In this example, the component RenderName expects to receive a prop called name which is a string. If this prop is not passed, or if it is not a string, a warning will be displayed.
+
+### Using defaultProps
+
+Another cool thing we can do with PropTypes is passing in default props:
+
+~~~javascript
+import React from 'react';
+import PropTypes from 'prop-types';
+
+const RenderName = (props) => {
+  return <div>{props.name}</div>;
+};
+
+RenderName.propTypes = {
+  name: PropTypes.string,
+};
+
+RenderName.defaultProps = {
+  name: 'Zach',
+};
+
+export default RenderName;
+~~~
+
+In this example, with the help of the defaultProps property we are defining a default value for the `name` prop. This way, if the `RenderName` component is called without passing in the `name` prop, it will default to "Zach". When you do pass in props, they will take precedence over the default props.
+
+### What about TypeScript?
+
+Now is also a good time to mention [TypeScript](typescriptlang.org) - a strongly typed language that builds on JavaScript. We don't cover it specifically, but it's worth learning about it if you'd like more safety while writing your code.
+
+If you go in the direction of learning TypeScript, our recommendation would be picking up a previous project and refactoring the components one by one to TypeScript. Learn by doing!
```

Comment:
I'm slightly hesitant on providing a "how to learn TypeScript" recommendation as it _might_ hold us accountable for providing support (for TS) in our Discord chats. I'd like to hear your thoughts though

Reply:
@01zulfi I had the same doubts, however if we are bringing up TypeScript so learners are aware of it I feel uneasy at the very least not pointing them in the right direction. 

people often ask for advice on how to start with TS, and i always tell them the same thing. 

i would love if we did mention it this way. Perhaps adding a disclaimer how this is just a recommendation and that we don't explicitly cover or provide support for TS?

Reply:
I see what you're saying.

A couple of suggestions to improve this section:

- We could rephrase the "We don't cover it specifically," to "We don't cover it in our curriculum yet,"

- At this stage of the curriculum, learning TS can be a lot of overhead when learners are already learning React. Perhaps we can incorporate this into our recommendation:

> Learning TypeScript can be a lot of overhead when you're already learning React. Although, once you're comfortable navigating the ins and outs of React, you'll have a better time learning and applying TypeScript to your projects. If you go in the direction of learning TypeScript, our recommendation would be picking up a previous project and refactoring the components one by one to TypeScript. Learn by doing!

Reply:
@01zulfi done ! :rocket: